### ISSUE TITLE
Classification accuracy low

### ISSUE THREAD

**Issue by XingZeng307** [ultralytics/yolov5#3232]
## ❔Question

Hi, thanks for the great work and opening up the classifier branch. 

Could you please check again? I did training on MNIST too, using yolov5s, but only 41% accuracy in top5 epochs, but if using EfficientNet, the results will be similar to the above one you were shown in issue#2001. (MNIST is at 99% in 5 epochs.)

I also tried a different dataset, there is still a big difference in accuracy between yolov5 classifier and EfficientNet classifier.

I pulled the latest repository. (don't know why it doesn't work). Here is my result.

python classifier.py --data mnist
output:

```
epoch gpu_mem train_loss val_loss accuracy
1/100 0.493G 2 1.86 0.277 : 100%|| 469/469 [00:09<00:00, 49.32it/s]
2/100 0.95G 1.8 1.8 0.273 : 100%|| 469/469 [00:09<00:00, 49.34it/s]
3/100 0.95G 1.72 1.77 0.3 : 100%|| 469/469 [00:09<00:00, 48.96it/s]
4/100 0.95G 1.65 1.67 0.355 : 100%|| 469/469 [00:09<00:00, 49.04it/s]
5/100 0.95G 1.61 1.63 0.417 : 100%|| 469/469 [00:09<00:00, 48.98it/s]
6/100 0.95G 1.58 1.63 0.388 : 100%|| 469/469 [00:09<00:00, 48.68it/s]
7/100 0.95G 1.54 1.57 0.403 : 100%|| 469/469 [00:09<00:00, 48.65it/s]
8/100 0.95G 1.52 1.55 0.417 : 100%|| 469/469 [00:09<00:00, 48.64it/s]
9/100 0.95G 1.49 1.52 0.43 : 100%|| 469/469 [00:09<00:00, 48.50it/s]
10/100 0.95G 1.47 1.52 0.423 : 100%|| 469/469 [00:09<00:00, 48.28it/s]

```

Thank you!

**Comment by glenn-jocher** [comment]
@XingZeng307 classifier branch was behind by quite a few commits, so I merged master and re-ran the example usage, and also observe poor results now. I'll take a look at this this week, as YOLOv5s should be producing about 0.99 accuracy on MNIST also, so it looks like there's a bug somewhere. Thanks for letting us know!

<img width="1000" alt="Screenshot 2021-05-19 at 13 09 20" src="https://user-images.githubusercontent.com/26833433/118803257-70685d80-b8a3-11eb-8400-85da6a8e957a.png">

**Comment by glenn-jocher** [comment]
@XingZeng307 the output is shown here most recently on 10 March: https://github.com/ultralytics/yolov5/issues/2409#issuecomment-794591997

So it seems like something changed between then and now to cause the accuracy drop.

**Comment by glenn-jocher** [comment]
@XingZeng307 Ok I see the problem. In the new training only the new Classify() head has it's gradients turned on, so the entire backbone is frozen. You can see in my screenshot from today's classifier training it says `1192362 parameters, 2570 gradients`, whereas in the March 2021 results it shows `1194954 parameters, 1194954 gradients`.

**Comment by glenn-jocher** [comment]
@XingZeng307 good news 😃! Your original issue may now been fixed ✅ in https://github.com/ultralytics/yolov5/commit/ad50c810077c72429433d0294474f53d6d6cd80f.  To receive this update you can:
- `git pull` from within your `yolov5/` directory
- `git clone https://github.com/ultralytics/yolov5` again
- Force-reload [PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/): `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- View our updated notebooks:  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by XingZeng307** [comment]
@glenn-jocher  Thank you sooo much. It works now!!

**Comment by glenn-jocher** [comment]
@XingZeng307 no problem. The classifier branch still needs some work to figure out the best hyperparameters to use for training though. I noticed if I set the epochs too short, i.e. --epochs 5 the losses become unstable, but at --epochs 10 they train fine.

**Comment by XingZeng307** [comment]
@glenn-jocher  Yeah. As you did in detection, maybe applying hyperparameter evolution helps for hyperparameters optimization? And also adapting the mosaic dataloader for better performance.

**Comment by mzhadigerov** [comment]
Do I need to clone from 'classifier' branch or from the master/main branch?? Since there is no `classifier.py` in the main branch

**Comment by mzhadigerov** [comment]
I'm also getting very low accuracy on the custom dataset. 

I'm using this command: `!python classifier.py --data orientation --epochs 5 --img 640 --batch 16 --model yolov5s --cache` 

I'm getting around 40% accuracy.
